cmd/syncthing/gui.go

```diff
@@ -1567,15 +1567,51 @@ func browseFiles(current string, fsType fs.FilesystemType) []string {
 
 	fs := fs.NewFilesystem(fsType, searchDir)
 
-	subdirectories, _ := fs.Glob(searchFile + "*")
+	subdirectories, _ := fs.DirNames("")
 
-	ret := make([]string, 0, len(subdirectories))
+	type match struct {
+		path             string
+		case_differences int
+	}
+
+	matches := make([]match, 0, len(subdirectories))
+
+next_subdir:
 	for _, subdirectory := range subdirectories {
+		var m match
+		for i := 0; i < len(subdirectory) && i < len(searchFile); i += 1 {
+			if subdirectory[i] == searchFile[i] {
+				continue
+			}
+
+			if strings.EqualFold(string(subdirectory[i]), string(searchFile[i])) {
```

Comment:
This can be funky with utf8 where a character is more than one byte, perhaps you should extract runes using utf8 package, and do unicode.ToLower(rune) etc.